src/sort.rs

```diff
@@ -121,11 +122,12 @@ where
             let pivot_index = rng.gen_range(0, n);
             let partition_index = self.partition_mut(pivot_index);
             if i < partition_index {
-                self.slice_mut(s![..partition_index]).get_from_sorted_mut(i)
+                self.slice_axis_mut(Axis(0), Slice::from(..partition_index))
+                    .get_from_sorted_mut(i)
             } else if i == partition_index {
                 self[i].clone()
             } else {
-                self.slice_mut(s![partition_index + 1..])
+                self.slice_axis_mut(Axis(0), Slice::from(partition_index + 1..))
```

Comment:
I can see the rationale for all the changes you have proposed, but I feel I am missing something here: why is `slice_axis_mut` preferable to `slice_mut`?

Reply:
It's just slightly lower overhead, that's all. `.slice_mut()` goes through all the n-dimensional slicing infrastructure and in the end [calls `.slice_axis_inplace()`](https://github.com/rust-ndarray/ndarray/blob/0.12.1/src/impl_methods.rs#L374); `.slice_axis_mut()` [jumps there directly](https://github.com/rust-ndarray/ndarray/blob/0.12.1/src/impl_methods.rs#L411). The performance difference is quite small in this case (~2% in the small array tests when I tried it earlier), so if `.slice_mut()` seems cleaner, that's fine too.

Reply:
I think

Reply:
Readibility-wise, I'd say they are the same. I was just curious to understand the rationale, your laser-focused optimization eye is something I am trying to learn from your PR review :stuck_out_tongue: